Add redis subchart

ArmaanT opened the pull request:
This PR:
* adds the ability to use the Bitnami Redis chart as a subchart so that renovate can be configured to use Redis for caching.
* Modifies the CI configuration to be able to handle subcharts 
* Adds an additional set of values with redis enabled for `ct` to lint and run tests on

Closes #113

viceice reviewed (commented):
Do we need to commit the subchart? I think this can be removed from repo.

viceice commented:
It needs a github workflow change to add the helm repo for bitnami charts

ArmaanT commented:
> Do we need to commit the subchart? I think this can be removed from repo.

I think it needs to be committed. If I delete the directory and try to helm install, I get the following error:

```
Release "renovate" does not exist. Installing it now.
Error: found in Chart.yaml, but missing in charts/ directory: redis
```

JamieMagee commented:
I was just about to post your edit 😅 We need to add

```sh
helm repo add bitnami https://charts.bitnami.com/bitnami
helm dependency build
```

viceice reviewed (commented):
Can you please try @JamieMagee plan.

ArmaanT commented:
Okay, this should be ready for review now. I've updated the original PR description with an explanation of my changes.

I haven't tested the release workflow changes. However, it uses very similar logic to kubeval, so I'm reasonably confident it will work.

viceice commented on README.md:
````diff
@@ -29,3 +29,13 @@ helm install --generate-name --set renovate.config='\{\"token\":\"...\"\}' renov
+When using this repo locally or contributing to this repo, you will need to build the dependencies used for each helm chart. You can run the following commands to do so:
+
+```bash
+cd charts/renovate
````
Are you sure this needs to be run in chart subdir instead of repo root?

ArmaanT replied:
The build command does, but not the repo add. Moved the order of those to make it more clear

viceice commented on .github/workflows/release.yaml:
```diff
@@ -5,7 +5,7 @@ on:
       - master
     paths:
-      - 'charts/**'
+      - "charts/**"
```
single quotes please

ArmaanT replied:
Fixed

viceice commented on .github/workflows/release.yaml:
```diff
@@ -26,9 +26,19 @@ jobs:
           version: v1.1.1 #renovate: datasource=github-releases depName=helm/chart-releaser
         env:
-          CR_TOKEN: '${{ secrets.CR_TOKEN }}'
+          CR_TOKEN: "${{ secrets.CR_TOKEN }}"
```
single quotes please

ArmaanT replied:
Fixed

ArmaanT commented on .github/kubeval.sh:
```diff
@@ -12,7 +12,11 @@ SCHEMA_LOCATION="https://raw.githubusercontent.com/instrumenta/kubernetes-json-s
 tar -xf /tmp/kubeval.tar.gz kubeval
 
+# add helm repos
+helm repo add bitname https://charts.bitnami.com/bitnami
```
nice catch

viceice reviewed (approved):
@JamieMagee what do you think?

JamieMagee commented on charts/renovate/values.yaml:
```diff
@@ -71,3 +71,21 @@ resources:
+
+redis:
+  # Configuration for a Redis subchart. Additional documenation at
+  # https://github.com/bitnami/charts/tree/master/bitnami/redis.
```
This is a super small nit, but the full stop here breaks this link for me

```suggestion
  # https://github.com/bitnami/charts/tree/master/bitnami/redis
```

JamieMagee commented:
Overall looks good. I just have 2 questions:
1. Is there any other configuration that a user might want to pass through to redis?
2. Is there any way to add the bitnami repo automatically? It's defined in `charts/renovate/Chart.yaml` and I would hope that `helm` could read that

ArmaanT commented:
1. Users will be able to configure redis using any of the values supported by the bitnami chart. I just added a few defaults that made sense to me. By using the `redis.fullname` template the chart should be able to handle any custom configuration passed to redis.
2. I don't think so unfortunately. I removed the bitnami repo locally and tried to run `helm dependency build` but it complains that there's no repository definition for https://charts.bitnami.com/bitnami. Seems like helm only uses the definition to make sure the user has a repo with that URL and can't directly install from it

JamieMagee commented:
Thanks for the answers. I had a look into 2 myself, and it appears that repositories are not added automatically due to security concerns.

LGTM
